src/test/kotlin/com/example/Employee/services/BankAccountServiceTest.kt

```diff
@@ -0,0 +1,144 @@
+import com.example.Employee.models.BankAccount
+import com.example.Employee.models.Employee
+import com.example.Employee.repositories.BankAccountRepository
+import com.example.Employee.repositories.EmployeeRepository
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.Test
+import java.util.*
+import org.amshove.kluent.`should be equal to`
+import org.amshove.kluent.invoking
+import org.amshove.kluent.shouldThrow
+
+
+internal class BankAccountServiceTest {
+    private val employeeRepository= mockk<EmployeeRepository>()
+    private val bankAccountRepository=mockk<BankAccountRepository>()
+    private val bankAccountService=BankAccountService(employeeRepository,bankAccountRepository)
+
+    @Test
+    fun `should create bank account for employee`(){
+        every { employeeRepository.findById(fakeEmployee.employeeId) } returns Optional.of(fakeEmployee)
+        every { bankAccountRepository.save(fakeBankAccount) } returns fakeBankAccount
+
+        bankAccountService.addAccount(fakeEmployee.employeeId, fakeBankAccountRequest)
+
+        verify { bankAccountRepository.save(fakeBankAccount) }
+    }
+
+    @Test
+    fun `should throw 404 while trying to create bankAccount for non existent employee`(){
+        every { employeeRepository.findById(fakeEmployee.employeeId) }returns Optional.empty()
+
+        invoking {bankAccountService.addAccount(fakeEmployee.employeeId, fakeBankAccountRequest)} shouldThrow EntityNotFoundException("employee not found")
+    }
+
+    @Test
+    fun `should get all bank accounts of an employee`(){
+        every { employeeRepository.findById(fakeEmployee.employeeId) } returns Optional.of(fakeEmployee)
+        every { bankAccountRepository.findAllByEmployeeEmployeeId(fakeEmployee.employeeId) }returns listOf(
+            fakeBankAccount)
+
+        val fetchedBankAccounts= bankAccountService.getAccounts(fakeEmployee.employeeId)
+
+        fetchedBankAccounts `should be equal to` listOf(fakeBankAccount)
+    }
+
+    @Test
+    fun `should throw 404 while trying to get bank accounts of non existent employee`(){
+        every { employeeRepository.findById(fakeEmployee.employeeId) } returns Optional.empty()
+
+        invoking { bankAccountService.getAccounts(fakeEmployee.employeeId) } shouldThrow EntityNotFoundException("employee not found")
+    }
+
+    @Test
+    fun `should get a particular bank account of an employee`(){
+        every { employeeRepository.findById(1) } returns Optional.of(fakeEmployee.copy(employeeId = 1))
+        every {bankAccountRepository.findByEmployeeEmployeeIdAndId(1, fakeBankAccount.id)} returns Optional.of(
+            fakeBankAccount)
+
+        val fetchedBankAccount = bankAccountService.getAccount(1, fakeBankAccount.id)
+
+        fetchedBankAccount `should be equal to` fakeBankAccount
+    }
+
+    @Test
+    fun `should throw 404 while trying to get bank account of non existent employee`(){
```

Comment:
How is this different from the case of `should throw 404 while trying to get bank accounts of non existent employee`?

Reply:
I've actually written two different methods. One for getting all the bank accounts of an employee(which is getAccounts) and one for getting a particular bank account of an employee( which is getAccount).
So this one should throw error while trying to get the "bank account" using its id and the other one throws error while trying to get all bank accounts.

Reply:
Ok. Got it.